docs: improve cookbook to share client session and modules with classes (#5810)
dagger/dagger: docs/current/cookbook.md

<Tabs groupId="language">
<TabItem value="Go">

```go title="main.go" file=./cookbook/snippets/modules-shared-client/functions/main.go
```

```go title="pipelines/pipelines.go" file=./cookbook/snippets/modules-shared-client/functions/pipelines/pipelines.go
```

</TabItem>
<TabItem value="Node.js">

```typescript title="index.mts" file=./cookbook/snippets/modules-shared-client/functions/index.mts
```

```typescript title="pipelines.mts" file=./cookbook/snippets/modules-shared-client/functions/pipelines.mts
```

</TabItem>
<TabItem value="Python">

```python title="main.py" file=./cookbook/snippets/modules-shared-client/functions/main.py
```

```python title="pipelines.py" file=./cookbook/snippets/modules-shared-client/functions/pipelines.py
```

</TabItem>
</Tabs>

> Another possible approach shown into the following listing is to use independent classes (or interface depending on the programming language) with public methods as function.

Reviewer: ```suggestion
Another possible approach is to use independent classes (or interfaces, depending on the programming language) with public methods as functions. The following code listing demonstrates this approach.
```

Reviewer: Suggest changing the section header to "Organize pipeline code into reusable modules or classes"

> This is useful to avoid passing the client to all imported functions.

Reviewer: ```suggestion
```
